### DISCUSSION TITLE
Recompile modified DAFoam code in Docker Engine

### DISCUSSION THREAD

**Discussion by leocotteleer** [mdolab/dafoam#378] [Installation]
Dear all,

I am looking for any references or help to re-compile DAFoam inside a docker engine container. 

Only the source codes of OF1812/ADF/ADR are available when I mount the container.. Is there any means to test built-in code based on DAFoam inside Docker ?
Should I first download the source code from GitHub, compile it inside the Docker image and then change the path?

Thanks in advance!

**Comment by friedenhe** [comment]
Have you tried this? https://dafoam.github.io/mydoc_installation_docker.html. You can modify the script to compile your implementation instead of recompiling the whole DAFoam.

**Reply by leocotteleer** [reply]
Hi @friedenhe, thanks for your answer!

Yes I've tried but I had some errors..

When I use the Dockerfile (https://dafoam.github.io/mydoc_installation_docker.html) and I simply execute
`docker  -t my_new_dafoam_image` 
I had a _permission denied_ for the _mkdir -p /dafoam/repos_ command.

So, I tried to switch to root and chown the folder as:
`USER root &&
RUN mkdir -p $DAFOAM_ROOT_PATH/repos &&
RUN chown dafoamuser $DAFOAM_ROOT_PATH/repos &&
USER dafoamuser`
It enables to create the folder.

But then I had the following error:
`[4/5] RUN cd /dafoam/repos &&     git clone https://github.com/mdolab/dafoam &&     cd dafoam &&     . /dafoam/OpenFOAM/OpenFOAM-v1812/etc/bashrc &&     ./Allmake &&     . /dafoam/OpenFOAM/OpenFOAM-v1812-ADR/etc/bashrc &&     ./Allclean &&     ./Allmake &&     . /dafoam/OpenFOAM/OpenFOAM-v1812-ADF/etc/bashrc &&     ./Allclean &&     ./Allmake &&     pip install .:
0.332 Cloning into 'dafoam'...
1.926 /bin/sh: 1: .: Can't open /dafoam/OpenFOAM/OpenFOAM-v1812/etc/bashrc`

I use docker version 20.10.13 on macOS Big Sur.

Thanks for your help!

**Reply by friedenhe** [reply]
No, you can not use the root account to do things in the Docker image. Can you use the docker image to run an optimization without the permission issue? Make sure you are not in the $HOME directory when loading/running the docker.
